modules/vault/src/main/java/org/apache/ignite/internal/vault/VaultManager.java

```diff
@@ -171,6 +175,31 @@ public boolean bootstrapped() {
         }
     }
 
+    /**
+     * Persist node name to the vault.
+     *
+     * @param name Node name to persist.
+     * @return Future representing pending completion of the operation. Couldn't be {@code null}.
+     */
+    @NotNull public CompletableFuture<Void> putName(@NotNull String name) {
+        return put(NODE_NAME, name.getBytes(StandardCharsets.UTF_8));
+    }
+
+    /**
+     * @return Node name, if was stored earlier.
+     * @throws IgniteInternalCheckedException If couldn't get node name from the vault.
+     */
+    public String name() throws IgniteInternalCheckedException {
+        synchronized (mux) {
+            try {
+                return new String(vaultService.get(NODE_NAME).get().value());
```

Comment:
I'm not sure, but should we consider some caching here? Seems that `name()` might be called frequently, WDYT?

Reply:
Maybe, not sure too :) `appliedRevision` can have the same issue, maybe we should fix it in the scope of another issue?